### PR TITLE
Replace the QIR ExecutionManager with direct CircuitSimulator integra…

### PR DESCRIPTION
**Pull request by amccaskey**
There is no need to jump through the `extern "C"` QIR function calls to get to the `CircuitSimulator` for library-mode execution. This PR therefore removes the `QIRExecutionManager` and replaces it with a new one with direct integration with the `CircuitSimulator`. 

This will be helpful for future `CircuitSimulator` work that exposes operations like `exp (i theta Pauli)` (#654).

### CODE REVIEW

**Review by bmhowe23 — APPROVED** [review]
LGTM 👍

**Review by 1tnguyen — APPROVED** [review]
👍

File: runtime/cudaq/qis/managers/default/DefaultExecutionManager.cpp [hunk 1]
```diff
@@ -0,0 +1,226 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2023 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+#include "llvm/ADT/StringSwitch.h"
+
```
Comment (schweitzpgi) [inline runtime/cudaq/qis/managers/default/DefaultExecutionManager.cpp:10]: nit: blank line

File: runtime/cudaq/qis/managers/default/DefaultExecutionManager.cpp [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+    // Change basis, flush the queue
+    if (!basisChange.empty()) {
+      for (auto &basis : basisChange)
+        basis(false);
+
+      simulator()->flushGateQueue();
+    }
+
+    // Get whether this is shots-based
+    int shots = 0;
```
Comment (schweitzpgi) [inline runtime/cudaq/qis/managers/default/DefaultExecutionManager.cpp:192]: nit: should shots be std::size_t?

**Pull request merged**